pkg/threescale/threescale_integration_test.go

```diff
@@ -50,95 +68,113 @@ func TestAuthorizationCheck(t *testing.T) {
 	defer backendServer.Close()
 
 	inputs := []struct {
+		name     string
 		callWith []integration.Call
 		expect   string
 	}{
 		{
+			name: "Test No Authn Credentials Provided Denied",
 			callWith: []integration.Call{
 				{
 					CallKind: integration.CHECK,
 					Attrs: map[string]interface{}{
-						"request.path":       "/thispath?user_key=VALID",
+						"request.path":       "/thispath",
 						"request.method":     "get",
 						"destination.labels": map[string]string{"service-mesh.3scale.net": "true"},
 					},
 				},
 			},
-			expect: `
-			{
-			    "AdapterState":null,
-			    "Returns":[
+			expect: generatedExpectedError(t, rpc.PERMISSION_DENIED, unauthenticatedErr),
+		},
+		{
+			name: "Test Mismatched Credentials Provided Denied",
+			callWith: []integration.Call{
 				{
-				    "Check":{
-					"Status":{},
-					"ValidDuration": 0,
-					"ValidUseCount": -1
-				    },
-				    "Quota":null,
-				    "Error":null
-				}
-			    ]
-			}`,
+					CallKind: integration.CHECK,
+					Attrs: map[string]interface{}{
+						"request.path":       "/thispath",
+						"request.method":     "get",
+						"request.headers":    map[string]string{"x-app-key": "MIS", "x-user-key": "MATCH"},
+						"destination.labels": map[string]string{"service-mesh.3scale.net": "true"},
+					},
+				},
+			},
+			expect: generatedExpectedError(t, rpc.PERMISSION_DENIED, authMismatchErr),
```

Comment:
I'd say we prefer this to be "I will use `user_key=MATCH` and ignore `app_key`".
We can define this order:
1. Presence of user_key? => process user_key, goto 3
2. Presence of app_id? => goto 2.a
  2.a Presence of app_key? => process app_id with app_key, goto 3
  2.b Process app_id without app_key
3. Done.

Of course, if we _knew_ we were supposed to only be using app_id/app_key, we'd be ignoring the other steps (ie. 1).